### ISSUE TITLE
When and how are new versions of dependencies chosen?

### ISSUE THREAD

**Issue by metaperl** [reahl/reahl#238]
1. watchdog is required to be between 0.8.3 and 0.8.3.999 which [3 years old](https://pypi.org/project/watchdog/#history)
1. selenium is required to be between 2.42 and 2.9999 which is [2 years old](https://pypi.org/project/selenium/#history)
1. twine is required to be between 1.11.0 and 1.11.999 which is [9 months old](https://pypi.org/project/twine/#history)
1. tzlocal requires a [2 years old version](https://pypi.org/project/tzlocal/#history)
1. webob requires a [3 year old version](https://pypi.org/project/WebOb/#history)

**Comment by IwanVosloo** [comment]
When we release a new version of Reahl, we normally try to update the versions of dependencies.
We do try to stick to semantic versioning conventions for Reahl itself and we treat the versions of dependencies as if they do too (which may or may not be true).

We also see dependencies as part of our programming interface when it comes to semantic versioning since someone using Reahl could potentially also be programming to the projects that Reahl depends on. Hence we should not go up a major version of a dependency if we do not go up a major version of Reahl based on the reasoning that such an upgrade may break someone's code.

So, in short we will be checking dependencies again before 4.1 is released and again before 5.0.

It would be nice if setuptools had a way for a local installation to override something like the version we set for a required project so that a local installation could force the installation of a higher version on a case-by-case basis...but I don't know whether something like that exists.
